lib/ansible/modules/cloud/amazon/route53_facts.py

```diff
@@ -193,25 +422,25 @@ def get_hosted_zone(client, module):
     else:
         module.fail_json(msg="Hosted Zone Id is required")
 
-    results = client.get_hosted_zone(**params)
-    return results
+    try:
+        hosted_zone = client.get_hosted_zone(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't get hosted zone")
+    hosted_zone_id = hosted_zone['HostedZone']['Id'].replace('/hostedzone/', '')
+    hosted_zone['HostedZone'].update(get_resource_tags(client, module, hosted_zone_id, 'hostedzone'))
+    return hosted_zone
 
 
 def reusable_delegation_set_details(client, module):
     params = dict()
-    if not module.params.get('delegation_set_id'):
-        if module.params.get('max_items'):
-            params['MaxItems'] = module.params.get('max_items')
-
-        if module.params.get('next_marker'):
-            params['Marker'] = module.params.get('next_marker')
-
-        results = client.list_reusable_delegation_sets(**params)
-    else:
+    if module.params.get('delegation_set_id'):
         params['DelegationSetId'] = module.params.get('delegation_set_id')
```

Comment:
I think 'DelegationSetId' needs to be 'Id'.

Reply:
Wow, I did not change this, how long has this bug been present! (Thanks again for spotting this)